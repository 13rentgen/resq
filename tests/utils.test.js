import {
    getElementType,
    buildNodeTree,
    findInTree,
} from '../src/utils'

import { vdom, tree, } from './__mocks__/vdom'

describe('utils', () => {
    it('getElementType', () => {
        expect(getElementType('test')).toBe('test')
        expect(getElementType(function testFn() {})).toBe('testFn')
    })

    it('buildNodeTree', () => {
        expect(buildNodeTree(vdom)).toMatchSnapshot()
    })

    it('findInTree', () => {
        let results = [tree,]
        const selectors = ['TestWrapper', 'div',]

        selectors.forEach((selector) => {
            results = findInTree(results, child => child.name === selector)
        })

        expect(results.length).toBe(2)
        expect(results).toMatchSnapshot()
    })
})
