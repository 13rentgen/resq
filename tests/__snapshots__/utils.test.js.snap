// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils buildNodeTree 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "name": "div",
          "node": <div />,
          "props": Object {},
          "state": Object {},
        },
        Object {
          "children": Array [],
          "name": "span",
          "node": <span />,
          "props": Object {},
          "state": Object {},
        },
        Object {
          "children": Array [],
          "name": "div",
          "node": <div />,
          "props": Object {},
          "state": Object {},
        },
      ],
      "name": "TestWrapper",
      "node": <div />,
      "props": Object {
        "myProps": "test prop",
      },
      "state": Object {
        "initialized": true,
      },
    },
  ],
  "name": undefined,
  "node": undefined,
  "props": Object {},
  "state": Object {},
}
`;

exports[`utils findInTree 1`] = `
Array [
  Object {
    "name": "div",
    "node": <div />,
    "props": Object {},
    "state": Object {},
  },
  Object {
    "name": "div",
    "node": <div />,
    "props": Object {},
    "state": Object {},
  },
]
`;
